drivers/hmis_external_apis/app/jobs/hmis_external_apis/publish_external_forms_job.rb

```diff
@@ -43,32 +43,9 @@ def data_source_id
 
   # construct custom data element definitions from the nodes in the form definition
   def update_cdeds(definition)
-    owner_type = HmisExternalApis::ExternalForms::FormSubmission.sti_name
-    cded_by_key = definition.custom_data_element_definitions.index_by(&:key)
-    missing = []
-    definition.walk_definition_nodes do |node|
-      cded_key = node['link_id']
-      attrs = nil
-      case node['type']
-      when 'STRING', 'CHOICE', 'BOOLEAN'
-        # treat everything as a string for now
-        attrs = {
-          owner_type: owner_type,
-          field_type: 'string',
-          key: cded_key,
-          label: cded_key.humanize,
-          repeats: false,
-          UserID: user_id,
-          data_source_id: data_source_id,
-          form_definition_identifier: definition.identifier,
-        }
-      end
-      if attrs
-        cded = cded_by_key[cded_key]
-        cded ? cded.update!(attrs) : missing.push(attrs)
-      end
+    Hmis::Hud::CustomDataElementDefinition.transaction do
+      definition.introspect_custom_data_element_definitions(set_definition_identifier: true).each(&:save!)
```

Comment:
Do we need to keep track of `seen_keys` here the way that we do in `setup.rake`? I was thinking it's not needed because this is only 1 form's worth of CDEDs, so we won't see duplicates across forms, but let me know if that is incorrect -- I know we may add more public forms later. Would it be worthwhile to extract that logic into a reusable place? Maybe it would live as another method on `Form::Definition` that calls `introspect_custom_data_element_definitions` internally?